src/Microsoft.AspNetCore.Mvc.Testing/WebApplicationFactory.cs

```diff
@@ -327,6 +336,24 @@ public HttpClient CreateDefaultClient(Uri baseAddress, params DelegatingHandler[
         /// <inheritdoc />
         public void Dispose()
         {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        /// <param name="disposing">
+        /// <see langword="true" /> to release both managed and unmanaged resources;
+        /// <see langword="false" /> to release only unmanaged resources.
+        /// </param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
             foreach (var client in _clients)
```

Comment:
Wrap this code in an `if (disposing == true)` block please

Reply:
There is a roslyn codefix that will implement dispose using the right pattern for you. Just remove all the dispose implementation, ctrl+. in the IDisposable instance and choose implement IDisposable with disposing pattern or something like that.